Assets/Scripts/SceneManageMent/ScoreTracker.cs

```diff
@@ -133,18 +133,35 @@ public void AddToScore(int points)
     /// <summary>Updates this class's Energy to the bike's energy.</summary>
     private void UpdateUIEnergy()
     {
-        Energy = bike.Energy;
+        if (bike == null)
+        {
+            Debug.Log("There is no bike.");
+        }
+        else
+        {
+            Energy = bike.Energy;
+        }
+
     }
 
     /// <summary>Loads the gameover screen.</summary>
     /// <param name="survivedEvent">True if the player beat the level.</param>
     private void EndGame(bool survivedEvent)
     {
         PlayerDataObject.survivedEvent = survivedEvent;
+DLevel dLevel = Object.FindObjectOfType<DLevel>();
+        if (dLevel == null)
+        {
+            Debug.Log("cannot find Dlevel Object in scene.");
+        }
+        else
+        {
+            Object.FindObjectOfType<DLevel>().dangerTimer.Dispose(); //Dispose of timer for spawning more enemies
```

Comment:
Could you use the previously found "dLevel" variable?
```suggestion
            dLevel.dangerTimer.Dispose(); //Dispose of timer for spawning more enemies
```